pkg/resourcemanager/webhook/highavailabilityconfig/handler.go

```diff
@@ -367,3 +446,46 @@ func (h *Handler) mutateTopologySpreadConstraints(
 		podTemplateSpec.Spec.TopologySpreadConstraints = append(filteredConstraints, constraints...)
 	}
 }
+
+func mutateAutoscalingReplicas(
+	log logr.Logger,
+	failureToleranceType *gardencorev1beta1.FailureToleranceType,
+	obj client.Object,
+	getMinReplicas func() *int32,
+	mutateMinReplicas func(*int32),
+	getMaxReplicas func() int32,
+	mutateMaxReplicas func(int32),
+) error {
+	// do not mutate replicas if they are set to 0 (HPA disabled case)
+	if pointer.Int32Deref(getMinReplicas(), 0) == 0 && getMaxReplicas() == 0 {
+		return nil
+	}
+
+	replicas := kutil.GetReplicaCount(failureToleranceType, obj.GetLabels()[resourcesv1alpha1.HighAvailabilityConfigType])
```

Comment:
This code is reused in `mutateReplicas()`: https://github.com/gardener/gardener/blob/e2c5326b66365b62646ca517db505995a3f3a589/pkg/resourcemanager/webhook/highavailabilityconfig/handler.go#L240-L252
you can extract it into a separate function and call that instead